src/Arcus.Messaging.Abstractions.ServiceBus/AzureServiceBusMessageContext.cs

```diff
@@ -139,5 +147,70 @@ public static AzureServiceBusMessageContext Create(
 
             return new AzureServiceBusMessageContext(jobId, entityType, receiver, message);
         }
+
+        /// <summary>
+        /// Completes the Azure Service Bus message on Azure. This will delete the message from the service.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">Thrown when the message handler was not initialized yet.</exception>
+        public async Task CompleteMessageAsync(CancellationToken cancellationToken)
```

Comment:
Shouldn't these methods be declared as virtual or abstract methods on the `MessageContext` instead ?
I think these describe common functionality, whether you're working with Azure ServiceBus or any other message-broker, they describe some common messaging concepts imho.   Specific MessageContext implementations can then override / implement these methods.

I'm also thinking in terms of testability.  Suppose I want to create a unit-test for a certain MessageHandler, and I want to verify if the `DeadLetterMessageAsync` method is called in a certain situation, will I be able to do that when this is a 'concrete' method.  Can I easily mock/stub this ?

Reply:
I see.  One could argue that in those cases , the implementation of these methods should throw a NotSupportedException.  But, since we currently solely support Azure Service Bus, we can also consider this as a YAGNI atm.

For testing purposes, I like to use Moq and should we have an interface that defines these methods, it's easier -imho- to use that in tests instead of having to provide in-mem ServiceBusReceivers  etc

Reply:
Since these are `virtual`, `Moq`, could be used. But, for complexer scenario's, with actual Mock-verification, I would always recommend a dedicated implementation. 🙂

Reply:
Can we resolve this conv, or do we need to discus something else?

Reply:
I meant on the `ServiceBusReceiver`. It also has a empty `protrcted` constructor just fir this testing purpose https://learn.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebusreceiver.completemessageasync?view=azure-dotnet

Reply:
I can live with it for now.